### PR TITLE
Address test failures following CORS changes

### PR DESCRIPTION
**Pull request by lucafavatellawork**
Fixes https://github.com/opensparkl/sse_cli/issues/96

TODO:
- [x] Understand why websocket error when running sse_cli tests.
- [x] Check CI passes.

### CODE REVIEW

**Comment by lucafavatellawork** [comment]
Failure:
```
sparkl_cli/test/test_call.py ....
sparkl_cli/test/test_cd.py ..
sparkl_cli/test/test_cert.py ..WARNING: --verify is off, server validation disabled
.WARNING: --verify is off, server validation disabled
........WARNING: --verify is off, server validation disabled
.
sparkl_cli/test/test_common.py ...........
sparkl_cli/test/test_connect_close.py ..
sparkl_cli/test/test_listen.py .
sparkl_cli/test/test_log.py ..............
sparkl_cli/test/test_login_logout.py ...
sparkl_cli/test/test_password.py ......
sparkl_cli/test/test_put_source_rm.py .......Updated /private/var/folders/5k/8q0ml7wn0zl2v9c_1rqy9wzmp92w2k/T/pytest-of-luca.favatella/pytest-74/test_render_local_monitor_conn0/render_target.tmp.
Stop monitoring.
..
sparkl_cli/test/test_service.py F

================================================================================================= FAILURES =================================================================================================
__________________________________________________________________________________ test_sync_impl[connect_sparkl_setup0] ___________________________________________________________________________________

setup_test_service = ('auto_use_setup', 'pytest_nosession', 'test@localhost')

    def test_sync_impl(setup_test_service):
        session_alias, _no_session_alias, _test_user = setup_test_service
        # Load the external service implementation module.
        service = sparkl(
            "service",
            "TestRest/REST",
            TEST_MOD,
>           alias=session_alias)

sparkl_cli/test/test_service.py:96: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sparkl_cli/main.py:275: in sparkl
    return args.fun(args)
sparkl_cli/cmd_service.py:79: in command
    service = Service(args, module)
sparkl_cli/Service.py:60: in __init__
    self.__open(args)
sparkl_cli/Service.py:76: in __open
    self.ws = get_websocket(args, ws_path)
sparkl_cli/common.py:486: in get_websocket
    ws.connect(ws_url, cookie=cookie)
../../../homebrew/lib/python3.7/site-packages/websocket/_core.py:226: in connect
    self.handshake_response = handshake(self.sock, *addrs, **options)
../../../homebrew/lib/python3.7/site-packages/websocket/_handshake.py:80: in handshake
    status, resp = _get_resp_headers(sock)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

sock = <socket.socket [closed] fd=-1, family=AddressFamily.AF_INET6, type=SocketKind.SOCK_STREAM, proto=6>
success_statuses = (<HTTPStatus.MOVED_PERMANENTLY: 301>, <HTTPStatus.FOUND: 302>, <HTTPStatus.SEE_OTHER: 303>, <HTTPStatus.SWITCHING_PROTOCOLS: 101>)

    def _get_resp_headers(sock, success_statuses=SUCCESS_STATUSES):
        status, resp_headers, status_message = read_headers(sock)
        if status not in success_statuses:
>           raise WebSocketBadStatusException("Handshake status %d %s", status, status_message, resp_headers)
E           websocket._exceptions.WebSocketBadStatusException: Handshake status 400 Bad Request

../../../homebrew/lib/python3.7/site-packages/websocket/_handshake.py:165: WebSocketBadStatusException
========================================================================================= short test summary info ==========================================================================================
FAILED sparkl_cli/test/test_service.py::test_sync_impl[connect_sparkl_setup0] - websocket._exceptions.WebSocketBadStatusException: Handshake status 400 Bad Request
```

File: apps/sse_yaws/src/sse_yaws_util.erl [hunk 1]
```diff
@@ -195,7 +195,7 @@ response(Arg, {error, Reason}) ->
   Content =
     struct_to_content(Arg, Struct),
   [
-    {status, ?HTTP_OK},
+    {status, http_status(Reason)},
```
Comment (jacobythwaites) [inline apps/sse_yaws/src/sse_yaws_util.erl:198]: Ah, no. I wanted to get rid of this and have all generated responses have 200 OK

File: apps/svc_rest/ct_test/svc_rest_SUITE.erl [hunk 2]
```diff
@@ -202,7 +202,7 @@ test_post_service_not_client(_Config) ->
     httpc:request(post, Request, [], []),
   ct:pal("Response: ~p", [Response]),
 
-  {ok, {{"HTTP/1.1",200,"OK"}, _Headers, ResponseBody}} =
+  {ok, {{"HTTP/1.1",400,"Bad Request"}, _Headers, ResponseBody}} =
```
Comment (jacobythwaites) [inline apps/svc_rest/ct_test/svc_rest_SUITE.erl:205]: No, per above. The 200 OK was deliberate.

File: apps/svc_rest/ct_test/svc_rest_SUITE.erl [hunk 3]
```diff
@@ -343,7 +343,7 @@ test_post_simple_invalid_content_type(_Config) ->
         "&term=an_atom"}, [], []),
   ct:pal("Response: ~p", [Response]),
 
-  {ok, {{"HTTP/1.1",200,"OK"}, _Headers, ResponseBody}} =
+  {ok, {{"HTTP/1.1",400,"Bad Request"}, _Headers, ResponseBody}} =
```
Comment (jacobythwaites) [inline apps/svc_rest/ct_test/svc_rest_SUITE.erl:346]: Ditto

File: apps/svc_rest/src/svc_rest_yapp.erl [hunk 4]
```diff
@@ -198,7 +198,7 @@ handle(Arg,
       Arg#arg.headers, ?ORIGIN),
 
   if
-    RequestOrigin =/= undefined andalso
+    RequestOrigin =:= undefined orelse
```
Comment (jacobythwaites) [inline apps/svc_rest/src/svc_rest_yapp.erl:201]: Perfect, thanks :)

File: apps/sse_yaws/src/sse_yaws_util.erl (hunk 1 above)
Comment (lucafavatellawork) [reply apps/sse_yaws/src/sse_yaws_util.erl:198]: I am trying to follow the impact of this - so e.g. `GET /sse_cfg/cert` for existing user without client cert would return 200?

Comment (lucafavatellawork) [reply apps/sse_yaws/src/sse_yaws_util.erl:198]: And `/sse/ready`?

Comment (lucafavatellawork) [reply apps/sse_yaws/src/sse_yaws_util.erl:198]: I plan not to address.

File: apps/svc_rest/ct_test/svc_rest_SUITE.erl (hunk 3 above)
Comment (lucafavatellawork) [reply apps/svc_rest/ct_test/svc_rest_SUITE.erl:346]: Same as above.

**Pull request merged**